client/src/main/java/io/pravega/client/stream/impl/Orderer.java

```diff
@@ -32,13 +33,13 @@ SegmentInputStream nextSegment(List<SegmentInputStream> segments) {
             return null;
         }
         for (int i = 0; i < segments.size(); i++) {
-            SegmentInputStream inputStream = segments.get(counter.incrementAndGet() % segments.size());
+            SegmentInputStream inputStream = segments.get(MathHelpers.abs(counter.incrementAndGet()) % segments.size());
```

Comment:
yes, overflow. You are correct we call this every time a client calls readNextEvent()